lib/puppet/provider/zfs/zfs.rb

```diff
@@ -80,5 +90,14 @@ def exists?
     end
   end
 
+
+  define_method(:zoned) do
```

Comment:
Is there a particular reason why we're using `#define_method("zoned")` over `def zoned` ?

Reply:
My guess is because the surrounding code was using it. I personally was trying to change the original contributor's code from #6114 as little as possible.

Reply:
w4m